cartography/intel/gcp/storage.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+from googleapiclient.discovery import HttpError
+
+from cartography.intel.gcp import compute
+from cartography.util import run_cleanup_job
+logger = logging.getLogger(__name__)
+
+
+def get_gcp_buckets(storage, project_id):
+    """
+    Returns a list of storage objects within some given project
+
+    :type storage: The GCP storage resource object
+    :param storage: The storage resource object created by googleapiclient.discovery.build()
+
+    :type project_id: str
+    :param project_id: The Google Project Id that you are retrieving buckets from
+
+    :rtype: Storage Object
+    :return: Storage response object
+    """
+    try:
+        req = storage.buckets().list(project=project_id)
+        res = req.execute()
+        return res
+    except HttpError as e:
+        reason = compute._get_error_reason(e)
+        if reason == 'invalid':
+            logger.debug(
+                (
+                    "The project %s is invalid - returned a 400 invalid error."
+                    "Full details: %s"
+                ),
+                project_id,
+                e,
+            )
+            return None
```

Comment:
We found a case where a project ID exists in GCP, but still hits this code path.  Can you please change the `logger.debug` to `logger.warn`, and have this `return {}` instead?  

Returning `None` causes transform to fail later on with 

```
File \"/cartography/intel/gcp/storage.py\", line 202, in sync_gcp_buckets
    bucket_list = transform_gcp_buckets(storage_res)
  File \"/cartography/intel/gcp/storage.py\", line 62, in transform_gcp_buckets
    for b in bucket_res.get('items', []):
AttributeError: 'NoneType' object has no attribute 'get'"
```

This way when we hit line 62, `bucket_res.get()` will be the same as `{}.get()` and not `None.get()`, so we won't get the above exception.

Reply:
Done! Sorry for that over-look on my part. Also I used `logger.warning` instead of `logger.warn` because the latter is [depreciated](https://stackoverflow.com/a/15655674/10981911)

Reply:
Any idea why it hit that code path despite having a valid `project_id`?

Reply:
The project exists, but running get_gcp_buckets() on it resulted in a 400 with reason="invalid".  I'm not sure why this is the case (we might have some odd permissions thing going on? idk), but we did get other buckets.